component/exporter_test.go

```diff
@@ -12,66 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package component
+// TODO: Move tests back to component package after config.*Settings are removed.
+
+package component_test
```

Comment:
the package name seems incorrect

Reply:
I mean now we have `component` and `component_test` defined in `component/`

Reply:
It is, you can read https://go-review.googlesource.com/c/go/+/420415 and https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language/31443271#31443271

It is allowed, and we used this to avoid circular dependency [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/obsreport/obsreporttest/obsreporttest_test.go#L15) as well.

You can also see that I have a TODO to revert this after we remove the "*Settings".

Reply:
Nice! Thanks for providing the details